recipes/type_safe/meta.yaml

```diff
@@ -0,0 +1,49 @@
+{% set name = "type_safe" %}
+{% set version = "0.2.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/foonathan/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 34d97123fb9bca04a333565c4a2498425d602ec0c759de4be1b8cfae77d05823
+
+build:
+  number: 0
+  ignore_run_exports_from:
+    # Since this package does not ship any compiled code, we do not need any
+    # compatibility with the compiler used at runtime.
+    - {{ compiler('cxx') }}
+
+requirements:
+  build:
+    - cmake
+    # type_safe is header-only but its cmake setup is looking for a compiler to
+    # build some test programs (that are not installed.)
+    - {{ compiler('cxx') }}
+    - make  # [not win]
+  host:
+    # ensure that type_safe does not vendor debug_assert
+    - debug_assert
+  run:
+    # anything that uses type_safe headers is also going to need the debug_assert headers
+    - debug_assert
```

Comment:
Admittedly it's a bit strange that a header-only library has runtime dependencies. However, this has the right effect, if anything adds `type_safe` as a host dependency, it is also adding debug_assert. If anything is using `type_safe` at runtime, e.g., to runtime-compile C++ code against it, it is also going to need `debug_assert.`